scripts/jest/test-framework-setup.js

```diff
@@ -58,3 +58,26 @@ env.afterEach(() => {
   expect(console.error).toBeReset();
   expect(console.error).toNotHaveBeenCalled();
 });
+
+function wrapDevMatcher(obj, name) {
+  const original = obj[name];
+  obj[name] = function devMatcher() {
+    try {
+      original.apply(this, arguments);
+    } catch (e) {
+      global.__hadDevFailures = e.stack;
+    }
+  };
+}
+
+const expectDev = function expectDev(actual) {
+  const expectation = expect(actual);
+  if (global.__suppressDevFailures) {
```

Comment:
I was trying to have the test run to completion and record whether all the expectDev assertions pass independent of whether the other assertions in the tests pass. That way we can distinguish "passing" from "passing except dev" from "failing" with a single run.

You're right that I'm not testing that warnings don't happen in prod. This is mostly designed for fiber tests where we'll incrementally match dev behavior but can focus on tests that have prod behavior differences, rather than long-term prod testing to make sure we don't warn in prod, but we can probably adapt this infra for that later.